--- newlib_memccpy_bug.c	2022-12-24 20:17:41.952807796 +0530
+++ newlib_memccpy_correct.c	2022-12-27 23:10:35.182572732 +0530
@@ -80,11 +80,11 @@
   return ptr;
 #else
   void *ptr = NULL;
-  char *dst = dst0;
-  const char *src = src0;
+  unsigned char *dst = dst0;
+  const unsigned char *src = src0;
   long *aligned_dst;
   const long *aligned_src;
-  char endchar = endchar0 & 0xff;
+  unsigned char endchar = endchar0 & 0xff;
 
   /* If the size is small, or either SRC or DST is unaligned,
      then punt into the byte copy loop.  This should be rare.  */
